### ISSUE TITLE
Compiling CorsixTH in Windows with Eclipse & MinGW?

### ISSUE THREAD

**Issue by J-Shep** [CorsixTH/CorsixTH#995]
Hi,

I would like to be able to use Eclipse with MinGW instead of Visual Studio to compile & debug CorsixTH's source code, in addition to using Eclipse as I've always done to do CorsixTH Lua scripting and debugging but there are a few errors preventing CorsixTH from being compiled with the Eclipse Kepler project generated by CMake (I'm using it with Eclipse Mars). 

If you have any ideas for what the causes of these errors could be please post them:

1. '**stricmp' was not declared in this scope** : sdl_core.cpp : line 42
1. **Invalid arguments** : th_lua_gfx.cpp : line 260, 270 & 288
  `Example: l_freetype_throw_error_code(L, pFont->initialise());`
1.  **Symbol 'FT_Err_Ok' could not be resolved** : th_lua_gfx.cpp : line 239
1.  **Type 'FT_Error' could not be resolved**: th_lua_gfx.cpp: line 237

**Comment by TheCycoONE** [comment]
*stricmp* is not standard and should be replaced. It seems that for whatever reason ```_MSC_VER``` is defined in your setup. It should not be if you are not using Microsoft's compiler, you can check your list of defines and remove it if it's in there.

The rest of the errors are freetype related. You will need to tell us what freetype version you're using, and maybe provide a dump of your cmake output. See the travis-ci script for how to enable verbose debugging from cmake.

**Comment by J-Shep** [comment]
@TheCycoONE Hi, thanks for helping me, I will try to make CorsixTH compile in Eclipse now. 

I was going to compile CorsixTH without Freetype support, I'm sure I disabled it in CMake.

**Comment by TheCycoONE** [comment]
It is possible that compiling without freetype is broken, I haven't tried that path.

**Comment by J-Shep** [comment]
@TheCycoONE MinGW has to use MSVCRT.DLL so I guess this C++ runtime library is what defines _MSC_VER. Do you have any ideas for why the existing preprocessor directives for dealing with *stricmp* isn't suitable for MinGW despite it using the same runtime library as Visual Studio?

```
#include "config.h"
#include "lua_sdl.h"
#include "th_lua.h"
#include <cstring>
#include <cstdio>
#ifndef _MSC_VER
#define stricmp strcasecmp
#else
#pragma warning (disable: 4996) // CRT deprecation
#endif
```

**Comment by J-Shep** [comment]
I guess MinGW must be unable to redefine **stricmp** with **strcasecmp***.

**Comment by J-Shep** [comment]
MinGW must still not support *strcasecmp*, there's a [Google groups discussion](https://groups.google.com/forum/#!topic/json-c/gef4VwmOcqY) mentioning this from 2014.

**Comment by TheCycoONE** [comment]
We will replace all stricmp / strcasecmp instances because we don't like to use non-standard C / C++ functions in this project. In the mean time I suggest using one of the supported compiler / operating system combinations until someone with more experience with mingw can solve the issues.

**Comment by J-Shep** [comment]
@TheCycoONE Would you mind if I make a CorsixTH *stricmp / strcasecmp* function now so that it at least no longer has to use these non standard C++ functions from third party libraries and can support MinGW compiling in Windows?

**Comment by TheCycoONE** [comment]
@Alberth289346 said he would do it on irc earlier. The ones in sdl_core don't need to be strcasecmp, they can be regular strcmp or string.compare because those strings are always lowercase and fixed in lua code.

The one in sound probably has to stay, unless we know that the archives always have the same names and we fixed them in lua.

**Comment by J-Shep** [comment]
Thanks @Alberth289346 and thanks @TheCycoONE for helping me today.

**Comment by Alberth289346** [comment]
fixed in commit b458f0650d30d4c12f1b41ae83d3aa76c3a38fa9
